### PR TITLE
WHAN-33 ResourceMedia supports request attributes 'widths' and 'sizes'

### PR DESCRIPTION
**Pull request by stefanseifert**
this is a follow-up PR for #13 which got closed accidentally
related ticket: https://wcm-io.atlassian.net/browse/WHAN-33

### CODE REVIEW

File: media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java [hunk 1]
```diff
@@ -70,8 +82,55 @@
...
+   * Example: "{@literal ['(max-width: 799px)', '(min-width: 800px)']}"<br>
+   * You have to define the same number of array items in all pictureSource* properties.
+   * Cannot be used together with image sizes.
+   */
+  @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
+  private Object[] pictureSourceMedia;
+
+  /**
+   * List of widths for the picture source elements.
+   * Example: "{@literal 479,719,959,1279,1439,1440}"<br>
```
Comment (stefanseifert) [inline media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java]: add info about support for ? suffixed for the width to mark an with as optional - see
https://wcm.io/handler/media/component-properties.html

File: media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java [hunk 2]
```diff
@@ -70,8 +82,55 @@
   @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
   private String cssClass;
 
+  /**
+   * Defines responsive rendition widths for image.
+   * To be used together with 'imageSizes' property. <br>
+   * Example: "{@literal 2560:false,1920:false,1280:false,640:false,320:false}" <br>
```
Comment (stefanseifert) [inline media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java]: proposal for a better solution: we should not introduce a new syntax to mark sitzes as required or not. we should use the same syntax as used for the "width" property as described here:
https://wcm.io/handler/media/component-properties.html
number with "?" suffix = optional, number without suffix = mandatory

**Comment by stefanseifert** [comment]
@mrozati  please see the new review comments, especially the first one with the new syntax for mandatory/optional numbers is important, as we cannot change this later on. we should make it consistent from the start.

File: media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java (hunk 2 above)
Comment (mrozati) [reply media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java]: good idea. I changed that and moved the parseWidths method to an internal util class.

**Pull request merged**